src/hardware/serial_terminal.rs

```diff
@@ -1,5 +1,65 @@
 use super::UsbBus;
+use crate::hardware::flash::FlashSettings;
+use crate::hardware::flash::Settings;
 use core::fmt::Write;
+use miniconf::{JsonCoreSlash, TreeKey};
+
+struct Context {
+    output: OutputBuffer,
+    flash: FlashSettings,
+}
+
+const ROOT_MENU: menu::Menu<Context> = menu::Menu {
+    label: "root",
+    items: &[
+        &menu::Item {
+            command: "reset",
+            help: Some("Reset Stabilizer to force new settings to take effect."),
+            item_type: menu::ItemType::Callback {
+                function: handle_reset,
+                parameters: &[]
+            },
+        },
+        &menu::Item {
+            command: "list",
+            help: Some("List all available properties."),
+            item_type: menu::ItemType::Callback {
+                function: handle_list,
+                parameters: &[],
+            },
+        },
+        &menu::Item {
+            command: "read",
+            help: Some("Read a property from the device."),
+            item_type: menu::ItemType::Callback {
+                function: handle_property_read,
+                parameters: &[menu::Parameter::Optional {
+                    parameter_name: "property",
+                    help: Some("The name of the property to read. If not specified, all properties are read."),
+                }]
+            },
+        },
+        &menu::Item {
+            command: "write",
+            help: Some("Read a property to the device."),
+            item_type: menu::ItemType::Callback {
+                function: handle_property_write,
+                parameters: &[
+                    menu::Parameter::Mandatory {
+                        parameter_name: "property",
+                        help: Some("The name of the property to write."),
+                    },
+                    menu::Parameter::Mandatory {
+                        parameter_name: "value",
+                        help: Some("Specifies the value to be written to the property."),
```

Comment:
... `JSON` encoded.

Reply:
Fixed in https://github.com/quartiq/stabilizer/pull/805/commits/6f171975624442b05a68dee490f8802ecc7af40a